src/utils.rs

```diff
@@ -175,6 +176,46 @@ macro_rules! codegen_ret_extr {
 }
 pub(crate) use codegen_ret_extr;
 
+pub fn register_runtime_symbols(engine: &ExecutionEngine) {
+    #[cfg(feature = "with-runtime")]
+    unsafe {
```

Comment:
I think this function may be better as suggested below because if the `with-runtime` feature is disabled it doesn't make sense to have a `register_runtime_symbols` function.

```suggestion
#[cfg(feature = "with-runtime")]
pub fn register_runtime_symbols(engine: &ExecutionEngine) {
    unsafe {
```

Changing this will require adding `#[cfg(features = "with-runtime")]` to the function's invocations.